safehttp/plugins/hsts/hsts.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package hsts
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/google/go-safeweb/safehttp"
+)
+
+// Plugin implements automatic HSTS functionality.
+type Plugin struct {
+	maxAge            uint64
+	includeSubDomains bool
+	preload           bool
+}
+
+// NewPlugin creates a new HSTS plugin with safe defaults.
+func NewPlugin() Plugin {
+	return Plugin{
+		maxAge:            63072000, // two years in seconds
+		includeSubDomains: true,
+		preload:           false,
+	}
+}
+
+// Before should be executed before the request is sent to the handler.
+// The function redirects HTTP requests to HTTPS. When HTTPS traffic
+// is received the Strict-Transport-Security header is applied to the
+// response.
+func (p *Plugin) Before(w safehttp.ResponseWriter, r *safehttp.IncomingRequest) safehttp.Result {
+	if r.TLS == nil {
+		r.URL.Scheme = "https"
+		return w.Redirect(r, r.URL.String(), 301)
+	}
+
+	value := strings.Builder{}
```

Comment:
Nit: when initializing zero values it is preferred to use var.
```suggestion
	var value strings.Builder
```